### PR TITLE
ENG-2700: Clear up ContractID structs

### PR DESCRIPTION
**Pull request by SmartMonkey-git**
[Task](https://datarevenue.atlassian.net/browse/ENG-2700)
It was more confusing, then actually helping. Documentation has been inherited.

### CODE REVIEW

File: tycho-indexer/src/extractor/evm/mod.rs [hunk 1]
```diff
@@ -490,8 +490,11 @@ impl TvlChange {
 /// smart contract.
 #[derive(Debug, Clone, PartialEq, Default, Deserialize, Serialize)]
 pub struct ProtocolComponent {
-    // an id for this component, could be hex repr of contract address
-    pub id: ContractId,
+    /// Represents the unique identifier of a contract. It can represent an on-chain
+    /// address or in the case of a one-to-many relationship it could be something like
+    /// 'USDC-ETH'. This is for example the case with ambient, where one component is
+    /// responsible for multiple contracts.
```
Comment (dianacarvalho1) [inline tycho-indexer/src/extractor/evm/mod.rs]: is it though??? didn't we decide to have multiple components for each of the ambient pools (token pair) anyway. ai I feel like we keep having this discussion. maybe I'm confused too. what did we decide before? @tamaralipows

Comment (tamaralipows) [reply tycho-indexer/src/extractor/evm/mod.rs]: Yeah we decided to have multiple components for each of the pools (I think you told me that the most recently) haha. That's how I set it up in substreams at least.

Comment (dianacarvalho1) [reply tycho-indexer/src/extractor/evm/mod.rs]: @SmartMonkey-git can you just fix the docstring so it isn't misleading please?

Comment (kayibal) [reply tycho-indexer/src/extractor/evm/mod.rs]: I don't get why we would have multiple components per pool? I understand why we would have multiple components per contract.

**Pull request merged**